### PR TITLE
20 triplet lists

### PR DESCRIPTION
**Pull request by chcltchunk**
C function to generate a triplet list that references pairs in a given neighbour list.
Wiki entry as well as an application of the triplet list will follow soon.

### CODE REVIEW

**Comment by chcltchunk** [comment]
The test protocol is in a different file. I will add it tonight.

File: .gitignore [hunk 1]
```diff
@@ -58,4 +58,7 @@ docs/_build/
 *~
 
 #MAC DS_Store
-.DS_Store
\ No newline at end of file
+.DS_Store
+
+# everything else
+test_triplets.py
```
Comment (pastewka) [inline .gitignore:64]: This should not be checked into the repository.

File: c/matscipymodule.c [hunk 2]
```diff
@@ -51,6 +51,10 @@ static PyMethodDef module_methods[] = {
       "Compute a neighbour list for an atomic configuration." },
     { "first_neighbours", (PyCFunction) py_first_neighbours, METH_VARARGS,
       "Compute indices of first neighbours in neighbour list array." },
+    { "triplet_list", (PyCFunction) py_triplet_list, METH_VARARGS,
+      "Compute a triplet list for a first_neighbour list." },
+    { "get_jump_indicies", (PyCFunction) py_get_jump_indicies, METH_VARARGS,
```
Comment (pastewka) [inline c/matscipymodule.c:56]: What is the difference between `first_neighbours` and `get_jump_indices`?

File: c/neighbours.c [hunk 3]
```diff
@@ -691,3 +692,169 @@ py_first_neighbours(PyObject *self, PyObject *args)
...
+    }
+
+
+    npy_int *fi = NULL, *ij_t = NULL, *ik_t = NULL, *jk_t = NULL;
+
+    py_fi = PyArray_FROMANY(py_fi, NPY_INT, 1, 1, NPY_C_CONTIGUOUS);
+    fi = PyArray_DATA((PyArrayObject *) py_fi);
+
+    if (!fi) {
+        Py_XDECREF(fi);
```
Comment (pastewka) [inline c/neighbours.c:718]: Why the DECREF? fi must be NULL here.

File: c/neighbours.c [hunk 4]
```diff
@@ -691,3 +692,169 @@ py_first_neighbours(PyObject *self, PyObject *args)
...
+
+    double cutoff = DBL_MAX;
+
+    if (py_cutoff || py_absdist) {
+	if (!py_absdist || !py_cutoff) {
+	    PyErr_SetString(PyExc_TypeError, "Cutoff and distances must "
+                                                "be specified together.");
+	    return NULL;
+	}
+        py_absdist = PyArray_FROMANY(py_absdist, NPY_DOUBLE,
```
Comment (pastewka) [inline c/neighbours.c:730]: Indentation here and below

File: matscipy/calculators/__init__.py [hunk 5]
```diff
@@ -25,6 +25,7 @@
 from .pair_potential import PairPotential
 from .supercell_calculator import SupercellCalculator
 from .polydisperse import Polydisperse
+from .AbellTersoffBrenner import AbellTersoffBrenner
```
Comment (pastewka) [inline matscipy/calculators/__init__.py:28]: Please use underscores for module name, not CamelCase

Comment (pastewka) [reply matscipy/calculators/__init__.py:28]: i.e., this would be `abell_tersoff_brenner`

Comment (pastewka) [reply matscipy/calculators/__init__.py:28]: Also, this model does not seem to be part of this PR

**Review by pastewka — CHANGES_REQUESTED** [review]
Looks good, only a few minor things to resolve

**Pull request merged**